pkg/storage/streams/store.go

```diff
@@ -167,7 +191,41 @@ func (s *streamStore) Get(ctx context.Context, path paths.Path) (
 			return nil, Meta{}, err
 		}
 
-		rangers = append(rangers, rangeCloser)
+		decrypter, err := eestream.NewAESGCMDecrypter(&encKey, &firstNonce, s.encryptionBlockSize)
+		if err != nil {
+			for _, ranger := range rangers {
```

Comment:
maybe we could define a little cleanup helper func we could call, instead of having to copy paste this

Reply:
I don't know what happened to the other comment I made about this, but would it be possible to `defer _= ranger.Close()` every time we make a new one? That way we wouldn't have to explicitly handle it for each error.

But I don't understand enough about Go to know whether that's a valid solution

Reply:
oh, yeah, kaloyan responded to that - the difficulty is we're returning the ranger and when we return it we don't want it to be closed. so we have to do something that happens *only* in the error cases. defer runs when the function exits no matter what.

you could do

```golang

var thingsToCloseOnError []io.Closer
errcleanup := func() {
  for _, c := range thingsToCloseOnError {
    _ = c.Close()
  }
}

...
thingsToCloseOnError = append(thingsToCloseOnError, ranger)
...
if err != nil {
  errcleanup()
  return err
}
```

or something

Reply:
Thanks! I will do this

Reply:
@jtolds What's the benefit of the function as a variable (so JavaScript) in this case instead of defining a regular `errcleanup()` function?